eslint issues in props validation

The CircleCI build failed due to eslint issues below..

```
'form' is missing in props validation          react/prop-types
```

fixed them by adding to type Props object or ignore the line in ComposedComponent.

Actually I cannot install using npm, it shows me some errors:
```
npm ERR! Unexpected token 
npm ERR!  in JSON at position 419017 while parsing '{
npm ERR!   "name": "react-schema-form",
npm ERR!   "vers'
```
but using yarn install, it works fine!

--- a/src/Array.js
+++ b/src/Array.js
@@ -97,8 +97,9 @@ class Array extends Component<Props, State> {
         };
     }
 
-    static getDerivedStateFromProps(props, state) {
-        const propsKey = props.form.key;
+    static getDerivedStateFromProps(props: Props, state) {
+        const { form } = props;
+        const propsKey = form.key;
         if (
             props.form &&
             propsKey === state.formKey &&

--- a/src/ComposedComponent.js
+++ b/src/ComposedComponent.js
@@ -20,11 +20,19 @@ const defaultValue = props => {
     return value;
 };
 
+type Props = {
+    errorText: any,
+    form: any,
+    showErrors: boolean,
+    localization: any,
+    onChange: any
+};
+
 const getDisplayName = WrappedComponent =>
     WrappedComponent.displayName || WrappedComponent.name || "Component";
 
 export default (ComposedComponent, defaultProps = {}) =>
-    class Composed extends React.Component {
+    class Composed extends React.Component<Props> {
         displayName = `ComposedComponent(${getDisplayName(ComposedComponent)})`;
 
         constructor(props) {
@@ -33,16 +41,11 @@ export default (ComposedComponent, defaultProps = {}) =>
             this.state = this.constructor.getDerivedStateFromProps(this.props);
         }
 
-        static getDerivedStateFromProps(nextProps) {
+        static getDerivedStateFromProps(nextProps: Props) {
             const { errorText, form, showErrors, localization } = nextProps;
             const getLocalizedString =
                 localization && localization.getLocalizedString;
             const value = defaultValue(nextProps);
-            const validationResult = utils.validate(
-                form,
-                value,
-                getLocalizedString
-            );
             if (!showErrors) {
                 return {
                     value,
@@ -50,13 +53,23 @@ export default (ComposedComponent, defaultProps = {}) =>
                     error: ""
                 };
             }
+
+            const validationResult = utils.validate(
+                form,
+                value,
+                getLocalizedString
+            );
+
+            const error = !validationResult.valid
+                ? validationResult.error
+                : undefined;
+
             return {
                 value,
                 valid: validationResult.valid,
                 error:
-                    (!validationResult.valid
-                        ? validationResult.error.message
-                        : null) || errorText
+                    (!validationResult.valid ? error.message : null) ||
+                    errorText
             };
         }
 

--- a/src/MultiSelect.js
+++ b/src/MultiSelect.js
@@ -44,7 +44,6 @@ const MenuProps = {
 type Props = {
     model: any,
     form: any,
-    key: any,
     onChangeValidate: any,
     classes: any,
     localization: Localization

--- a/src/MultiSelect.js
+++ b/src/MultiSelect.js
@@ -63,10 +63,10 @@ class MultiSelect extends Component<Props, State> {
     }
 
     static getDerivedStateFromProps(props: Props) {
-        if (props.model && props.form.key) {
+        const { model, form } = props;
+        if (model && form.key) {
             return {
-                currentValue:
-                    utils.getValueFromModel(props.model, props.form.key) || []
+                currentValue: utils.getValueFromModel(model, form.key) || []
             };
         }
         return null;

--- a/src/Select.js
+++ b/src/Select.js
@@ -28,13 +28,11 @@ class Select extends Component<Props, State> {
         };
     }
 
-    static getDerivedStateFromProps(props) {
-        if (props.model && props.form.key) {
+    static getDerivedStateFromProps(props: Props) {
+        const { form, model } = props;
+        if (model && form.key) {
             return {
-                currentValue: utils.getValueFromModel(
-                    props.model,
-                    props.form.key
-                )
+                currentValue: utils.getValueFromModel(model, form.key)
             };
         }
         return null;

--- a/src/Select.js
+++ b/src/Select.js
@@ -10,7 +10,6 @@ import type { Localization } from "./types";
 
 type Props = {
     model: any,
-    key: any,
     form: any,
     onChangeValidate: any,
     localization: Localization

